### PR TITLE
[nudge-a-palooza] Redirect users to feature upsell pages based on plan features and not eligibility warnings

### PR DESCRIPTION
**Pull request by adamziel**
This PR fixes a problem with a previous redirect logic: p2EDhh-tc-p2
tl;dr - business plan users who did not have an AT were unable to upload plugins and themes because they were redirected to the upsell page.

This behavior was caused by the fact that we redirected based on the presence of eligibility warnings like "upgrade to a business plan to use this feature". This logic turned out to be invalid since custom domain were also an eligibility warning. As specified in the discussion above, some timing issues may have been involved and sometimes it was OK so we got "lucky" during testing and released it.

This PR changes the approach so that we redirect based on plan features instead. It also extracts that logic into a separate file so it's easier to reason about.

Test plan:
1. Assign yourself to `customPluginAndThemeLandingPages` group of `nudgeAPalooza` test
1. On a free site, go to /themes and /plugins, click on "Upload" button and confirm you were redirected to an upsell page
1. On a premium site, go to /themes and /plugins, click on "Upload" button and confirm you were redirected to an upsell page
1. On a business site with no AT, go to /themes and /plugins, click on "Upload" button and confirm you can see a upload form with a warning saying that you should setup a domain first
1. On a business site with an AT, go to /themes and /plugins, click on "Upload" button and confirm you can see a upload form with no warnings (sometimes I get redirected to /wp-admin/plugin-install.php which is a correct behavior as well)

### CODE REVIEW

**Comment by alisterscott** [comment]
Tested this branch and it fixes #26458

Thanks

**Comment by alisterscott** [comment]
One thing I noticed about these featured upsell pages is when using site picker to switch to a site that shouldn't show the upsell feature page it still shows, for example

1. Visit plugins and upload for a free site and see: http://calypso.localhost:3000/feature/plugins/freesite.wordpress.com
2. Choose switch site to a business site - the upsell is still displayed even though it's a business site http://calypso.localhost:3000/feature/plugins/businessite.wordpress.com

@adamziel: would you like this raised as a separate issue?
